pkg/sql/catalog/systemschema/system.go

```diff
@@ -342,6 +342,17 @@ CREATE TABLE system.sqlliveness (
     expiration       DECIMAL NOT NULL,
   	FAMILY fam0_session_id_expiration (session_id, expiration)
 )`
+
+	MigrationsTableSchema = `
```

Comment:
Upon reflecting this a bit, I think we should take a step back. The main goals here are

a) allow scripting against the migration process, i.e. allow operators to determine what the migration status is
b) allow troubleshooting when a migration goes wrong

We're unlikely to be able to achieve b) using just a SQL table. However, for a) a SQL table is a good fit. So let's optimize this table for A and open an issue about B.

The table can thus become relatively simple. I think there should be one row per version that does work (and one for major versions) with timing information, name, and a status that the migration work function can update at will. For example an active row could look like this

```go
"20.2-13", "running", "VersionFooBar", "fooing all the bars", <start ts>, <end ts if finished>, <last error if previous attempt failed>
```

My suggestion is to split this out into a separate PR (the table is not needed for the basic harness and the basic version of the manager and testing it). I would be curious what @ajwerner thinks of our schema attempts here.

for b), I wonder if we can leverage loggers. Basically, we give the work function a logger that logs to an auxiliary file, and make it clear that this logger should be used liberally. Here I am interested in @knz's input.
Additionally, we could also open a net/trace span and hook that up to the context, but this will be less useful as we don't collect net/trace in debug.zip. But - it could be a nice auxiliary way to expose it that can come in handy on systems where access to logs is difficult to attain.

Reply:
#57445, let's carry the discussion over there.